app/scripts/lib/multichain-api/provider-request.js

```
@@ -13,16 +13,17 @@ export async function providerRequestHandler(
     return end(new Error('missing CAIP-25 endowment'));
   }
 
+  const chainId = scope.split(':')[1];
+
+  if (!chainId) {
+    return end(new Error('missing chainId'));
```

if we use the helper, we can throw the parsing error from the helper back